pkg/git/RepositoryManager.go

```diff
@@ -268,7 +272,17 @@ func (impl RepositoryManagerImpl) getStats(commit *object.Commit) (object.FileSt
 			return
 		}
 	}()
-	fs, err := commit.Stats()
+
+	// create context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(impl.configuration.CommitStatsTimeoutInSec)*time.Second)
+	defer cancel()
+	fs, err := commit.StatsContext(ctx)
+
+	// for timeout error, return empty result without error
+	if err == context.DeadlineExceeded || err == object.ErrCanceled {
```

Comment:
Will this condition be checked in parallel if the commitStats method stucks in infinite loop?

Reply:
provided context is with timeout. 
so if context is timed out, in that case err would be thrown by StatsContext method

Reply:
Missed one line, sorted